### ISSUE TITLE
da.mean crashing with numpy ndarray as input

### ISSUE THREAD

**Issue by mraspaud** [dask/dask#5049]
Hi,

The following code crashes when run interactively in a fresh virtualenv:
```python
import numpy as np
import dask.array as da
arr = np.array([0.28588], dtype=np.float32)
da.mean(arr)
```

The traceback says:
```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/a001673/daskbug/lib/python3.6/site-packages/dask/array/core.py", line 1114, in __repr__
    self.dtype,
  File "/home/a001673/daskbug/lib/python3.6/site-packages/dask/array/core.py", line 1044, in dtype
    return self._meta.dtype
AttributeError: 'dict' object has no attribute 'dtype'
```

Versions tested on:
 * numpy 1.16.4
 * dask 2.0.0
 * toolz 0.9.0
 * python 3.6.4 and 3.7.3

Works when installing dask 1.2.2 instead

**Comment by pentschev** [comment]
This is fixed in #5035. I now see that build failed, which I will check, but I just tested your example with that branch and it works.

**Comment by mraspaud** [comment]
Fantastic! Thanks!
